pkg/queue/concurrency_state.go

```diff
@@ -78,10 +77,43 @@ func ConcurrencyStateHandler(logger *zap.SugaredLogger, h http.Handler, pause, r
 			return
 		}
 
-		resume()
+		logger.Info("Requests increased from zero")
+		if err := resume(); err != nil {
+			logger.Errorf("Error handling resume request: %v", err)
+			panic(err)
```

Comment:
This panic won't actually do anything anymore as it's now situated directly in `ServeHTTP` where panics are usually caught by the wrapping server. We might need an explicit `os.Exit(1)` here.

Reply:
whoops, looks like I did it on the one above, but not here :facepalm: